tests/unit/test_utils.py

```diff
@@ -1016,6 +1017,54 @@ def test_uses_dualstack(self):
             'https://s3-accelerate.dualstack.amazonaws.com/foo/key.txt')
 
 
+class TestDeepMerge(unittest.TestCase):
+    def test_simple_merge(self):
```

Comment:
We should add tests for other types, particularly for lists so we can capture the existing behavior.